🔥[WIP][NOT READY FOR REVIEW]🐍[EXPERIMENT] 🐉 Use Ubuntu LTS as master image by default 🔥

dims opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

dims commented:
/test pull-kubernetes-e2e-gce
/test pull-kubernetes-e2e-gce-100-performance
/test pull-kubernetes-e2e-kind

dims commented:
/test pull-kubernetes-e2e-gce-device-plugin-gpu

dims commented:
/test pull-kubernetes-integration

dims commented:
/test pull-kubernetes-e2e-gce
/test pull-kubernetes-e2e-gce-device-plugin-gpu
/test pull-kubernetes-e2e-kind

dims commented:
/test pull-kubernetes-e2e-gce
/test pull-kubernetes-e2e-kind

dims commented:
/test pull-kubernetes-e2e-gce-100-performance

dims commented:
/test pull-kubernetes-e2e-gce 
/test pull-kubernetes-e2e-gce-device-plugin-gpu
/test pull-kubernetes-e2e-kind
/test pull-kubernetes-e2e-kind-ipv6

dims commented:
/test pull-kubernetes-e2e-gce-100-performance

dims commented:
/test pull-kubernetes-e2e-gce
/test pull-kubernetes-node-e2e-containerd

aojea commented:
/retest

aojea commented on cluster/gce/gci/configure.sh:
```diff
@@ -393,6 +401,59 @@ function load-docker-images {
+  apt-get install -y --no-install-recommends \
+    apt-transport-https \
+    ca-certificates \
+    socat \
```
this won't work if we don't install docker, I imagine that this is only for testing but is easy to forget :smile:

dims commented:
node-e2e had a flake - https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/87654/pull-kubernetes-node-e2e/1225265169563652096/

/test pull-kubernetes-node-e2e

dims commented:
3 flakes and the cross timed out. No bazel related interrupt/timeout/ooms. the PR for bazel shutdown is holding up!

dims commented:
/test all

dims committed:
Add gid to config.toml only when docker group is present

If we don't install docker and install just containerd apt packages,
there is no docker group. In this scenario, we should not add the gid to
config.toml

dims committed:
Install containerd package depending on CONTAINER_RUNTIME

dims committed:
Ability to override versions of containerd/runc

dims commented:
/test pull-kubernetes-e2e-kind-ipv6

dims committed:
set CONTAINER_RUNTIME to containerd

dims commented:
/test pull-kubernetes-e2e-gce-ubuntu

dims commented:
/test pull-kubernetes-e2e-gce-ubuntu

dims commented:
This PR served us well. adios my friend. All the stuff here have gotten into k/k and k/test-infra resulting in:

https://testgrid.k8s.io/sig-release-master-informing#gce-ubuntu-master-default
https://testgrid.k8s.io/sig-release-master-informing#gce-ubuntu-master-containerd

/close
